homeassistant/helpers/entity.py

```diff
@@ -514,10 +514,6 @@ def _async_write_ha_state(self) -> None:
         if self.available:
             attr.update(self.state_attributes or {})
             extra_state_attributes = self.extra_state_attributes
-            # Backwards compatibility for "device_state_attributes" deprecated in 2021.4
-            # Add warning in 2021.6, remove in 2021.10
```

Comment:
Was there a warning anywhere?

Reply:
At least I was not able to find it :-(

Reply:
@emontnemery do you remember what we decided for this behavior? Is the comment just wrong or have we not carried out the plan?

Reply:
We forgot to add the warning. Let's do now and bump the removal to 2022.3 or so?